pkg/block/fetcher.go

```diff
@@ -529,6 +537,15 @@ const blockIDLabel = "__block_id"
 // Filter filters out blocks that have no labels after relabelling of each block external (Thanos) labels.
 func (f *LabelShardedMetaFilter) Filter(_ context.Context, metas map[ulid.ULID]*metadata.Meta, synced *extprom.TxGaugeVec, _ bool) error {
 	var lbls labels.Labels
+	supportedActions := map[relabel.Action]struct{}{relabel.Keep: {}, relabel.Drop: {}, relabel.HashMod: {}}
+
+	for _, cfg := range f.relabelConfig {
+		if _, isPresent := supportedActions[cfg.Action]; !isPresent {
+			level.Error(f.logger).Log("msg", "unsupported relabel action", "action", cfg.Action)
```

Comment:
I think this is too late for validation.. Why cannot we validate those during startup? (: And fail command totally (in `cmd/thanos/...`

Reply:
Thanks for the review.
Agreed, will update asap.

Reply:
Done.